### ISSUE TITLE
Asset URL causes Teepee to go into an infinite loop while expanding URL placeholders.

### ISSUE THREAD

**Issue by rustyconover** [assetgraph/assetgraph#1144]
Hi Team,

When encountering an asset URL like

`https://api.amplitude.com/httpapi?api_key=e102edba5e9caea6b89e3c04fac87a4d&event={%22event_type%22:%22Overall%20Active%20User%22,%22event_properties%22:{%22Email%20Send%20Time%22:%222019-05-10T10:43:42%22,%22Current%20Date%22:%2210/05/2019%22,%22UserId%22:5698327,%22campaign%22:%22Daily%20Digest%22,%22Platform%22:%22E-Mail%22},%22user_id%22:%225698327%22}`

Teepee goes into an infinite loop when it runs: [`Teepee.prototype.expandUrl`](https://github.com/One-com/teepee/blob/master/lib/Teepee.js#L254)

It seems that the `{}` are used for URL placeholders in Teepee.  Teepee doesn't appear to have really great docs (or really many at all).  Was there a reason it was chosen over bent or even request.js?

We could fix this by adding:

```js
assetConfig.url = assetConfig.url.replace(
              /[{}]/g,
            encodeURIComponent);
```

In the same area where control characters are escaped.

Rusty

**Comment by papandreou** [comment]
> Was there a reason it was chosen over bent or even request.js?

We used to use request.js, but it kept breaking in weird ways. I wrote teepee when I was working at One.com since we needed a functional http client in a lot of places. Back in 2015 I adopted it in assetgraph as well. Bent, got etc. didn't exist when the choice was made.

Now that it's 2021 it could be that it's time to switch to something else! 🧹 
As part of that plugin-ification effort I'm thinking of making it possible to move the http(s) support to a plugin as well. Maybe that's the right time to pick this up -- possibly in a competing plugin :)

For now I'll look into getting the bug fixed -- nice find!
